model-optimizer/extensions/ops/scatter.py

```diff
@@ -94,6 +100,33 @@ class ScatterElementsUpdate(Scatter):
     op = op_type = 'ScatterElementsUpdate'
     version = 'opset3'
 
+    @staticmethod
+    def infer(node: Node):
+        Scatter.infer(node)
+
+        node_name = node.soft_get('name', node.id)
+        connected_in_ports = [port for port in node.in_ports().values() if not port.disconnected()]
+        assert len(connected_in_ports) == 4, \
+            "Incorrect number of inputs for {} node".format(node_name)
+
+        input_value = node.in_port(0).data.get_value()
+        indices_value = node.in_port(1).data.get_value()
+        indices_shape = node.in_port(1).data.get_shape()
+        updates_value = node.in_port(2).data.get_value()
+        axis = node.in_port(3).data.get_value()
+        assert axis is not None and axis.size <= 1
```

Comment:
The axis can be None. This is a valid case so we should not assert! Instead the code should look like this:

```python
        if input_value is not None and indices_value is not None and updates_value is not None and axis is not None and axis is not None:
            assert axis.size == 1, "The node {} has axis input value size equal to {} but it should be exactly 1.".format(node_name, axis.size)
            axis = axis.item()
            out_value = input_value.copy()
            for idx in np.ndindex(*indices_shape):
                data_idx = list(idx)
                data_idx[axis] = indices_value[idx]
                out_value[tuple(data_idx)] = updates_value[idx]
            node.out_port(0).data.set_value(out_value)

```